app/controllers/api/v1/videos_controller.rb

```diff
@@ -0,0 +1,47 @@
+class Api::V1::VideosController < ApplicationController
+  before_action :check_admin, only: [:create, :update, :destroy]
+
+  def index
+    videos = Video.where(deleted_at: nil)
+    render json: videos, status: 200
+  end
+
+  def show
+    video = Video.find(params[:id])
+    render json: video, status: 200
+  end
+
+  def create
+    video = current_user.mcs_admin.videos.build(video_params)
+    if video.save
```

Comment:
@MykleNero line 16-20 is essentially the same thing as in the `UsersController` create method. I was thinking about making a method in `ApplicationController` called `attempt_save` and pass in the `video` or `user` object and try to DRY the code. It also applies to the `update` action. Maybe do a `attempt_update` as well. Let me know if you think thats a good idea.